site/app/templates/submission/homework/SubmitBox.twig

```diff
@@ -17,10 +17,56 @@
     <div class="upperinfo">
         <h1 class="upperinfo-left">New submission for: {{ gradeable_name }}</h1>
         {% if has_due_date %}
-            <h3 class="upperinfo-right">Due: {{ formatted_due_date }}</h3>
+            <h3 class="upperinfo-right">Due: {{ formatted_due_date }} <br>
+                {%  if due_in_a_day %}
```

Comment:
If the server time is only shown when the gradeable is due in a day, then it seems unnecessary to fetch server time continually when it is not displayed. I would suggest wrapping the script below in an `if` block like this one does.

Reply:
Makes sense, thanks @zjxiaohan! Will make that change!